_posts/2017-03-04-bringing-artsy-to-google-actions-assistant-home.markdown

```diff
@@ -0,0 +1,145 @@
+---
+layout: post
+title: "Bringing Artsy to Google Assistant \"Home\" with Actions"
+date: 2017-03-04
+comments: true
+author: db
+categories: [Google Home, Google Assistant, Google Actions]
+---
+Last month we [brought Artsy to Amazon Alexa](/2016/11/30/bringing-artsy-to-amazon-echo-alexa/). In this post we'll repeat that process and enable Artsy for Google Assistant, including Google Home.
+
+tl;dr You can try Artsy on your Google Home now, say "Ok Google, talk to Artsy".
+
+<center><img src='/images/2017-03-04-bringing-artsy-to-google-actions-assistant-home/google-home.png'></center>
+
+<!-- more -->
+
+### Conversation Actions
+
+Actions are two-way dialogs with users. A user invokes an action, which calls so-called _fulfillment_ code - a REST endpoint.
+
+If you've never worked with Google Actions, read the [conversation actions intro](https://developers.google.com/actions/develop/conversation).
+
+### Actions SDK
+
+To write an action get the [actions SDK](https://developers.google.com/actions/develop/sdk/getting-started), which is a node.js client library, describe the actions in JSON and use a command-line tool called `gactions` or a web simulator to invoke the action code.
+
+### Actions Server
+
+The actions SDK can be called from any node.js application, or with some help from [google-actions-server](https://github.com/manekinekko/google-actions-server) (GAS) via boilerplate code in the [google-actions-starter](https://github.com/manekinekko/google-actions-starter) library.
+
+An `action.json` that describes the action, including the invocation trigger or the text-to-speech voice.
+
+```json
+{
+  "versionLabel": "1.0.0",
+  "agentInfo": {
+    "languageCode": "en-US",
+    "projectId": "biesenbach-one",
+    "invocationNames": ["artsy"],
+    "voiceName": "female_1"
+  },
+  "actions": [{
+    "initialTrigger": {
+      "intent": "assistant.intent.action.MAIN"
+    },
+    "httpExecution": {
+      "url": "https://biesenbach-one.appspot-preview.com"
+    }
+  }]
+}
+```
+
+The action implementation lives in `lib/action.js` that imports and creates a new instance of `google-actions-server`.
+
+```js
+import { ActionServer } from '@manekinekko/google-actions-server';
+
+class ArtsyAction {
+  constructor() {
+    this.agent = new ActionServer();
+
+    this.agent.setGreetings([
+      `What is the name of the artist you would like to hear about?`
+    ]);
+  }
+
+  welcomeIntent(assistant) {
+    // implementation details
```

Comment:
It might be useful to add sample comment here about how to get input and the fact that we need to do regex or nlp at this point